app/controllers/teams_controller.rb

```diff
@@ -72,23 +76,14 @@ def team_params
     params.require(:team).permit(:name, :private, :description)
   end
 
-  def redirect_if_not_teammember_or_admin
-    return if team.teammember?(current_user.id) || current_user.admin?
-    flash[:error] = 'You are not member of this team'
-    redirect_to teams_path
-  end
-
-  def redirect_if_not_allowed_to_delete_team
-    return if can_delete_team?(team)
-    flash[:error] = t('flashes.teams.cannot_delete')
-    redirect_to teams_path
+  def team
+    @team ||= Team.find(params[:id])
   end
 
-  def can_delete_team?(_team)
-    current_user.admin?
-  end
+  def user_not_authorized(exception)
```

Comment:
why don't we use the ApplicationController#user_not_authorized here ?
maybe leave a comment or remove this